src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -178,6 +188,190 @@ protected boolean doCreateTableIfNotExists(
+                    asMap();
+
+            int currentSchemaVersion = tableSchema.get(schemaVersionField).
+                    getInt();
+
+            // If JSON schema version mismatch throw error
+            if (currentSchemaVersion != supportedSchemaVersion) {
+                String msg = String.format("Could not validate schema of the " +
+                                "table %s in the database and entity %s : " +
+                                "json schema version mismatch. " +
+                                "Expected version is %s but version is %s",
+                        entityInformation.getTableName(),
+                        entityInformation.getJavaType().getName(),
+                        currentSchemaVersion,
+                        supportedSchemaVersion);
+                throw new IllegalStateException(msg);
+            }
+
+            ArrayValue tableColumns = tableSchema.get(colField).asArray();
+            ArrayValue tableShardKeys =
+                    tableSchema.get(shardField).asArray();
+            ArrayValue tablePrimaryKeys =
+                    tableSchema.get(primaryField).asArray();
+
+            Map<String, String> tableShardMap = new LinkedHashMap<>();
+            Map<String, String> tableNonShardMap = new LinkedHashMap<>();
+            Map<String, String> tableOthersMap = new LinkedHashMap<>();
+
+            // extract table details into maps
+            for (int i = 0; i < tableColumns.size(); i++) {
+                MapValue column = tableColumns.get(i).asMap();
+                String colName =
+                        column.getString(colNameField).toLowerCase();
+                String columnType =
+                        column.getString(colTypeField).toLowerCase();
+
+                if (i < tableShardKeys.size()) {
+                    tableShardMap.put(colName, columnType);
+                } else if (i < tablePrimaryKeys.size()) {
+                    tableNonShardMap.put(colName, columnType);
+                } else {
+                    tableOthersMap.put(colName, columnType);
+                }
+            }
+
+            // extract entity details into maps
+            Map<String, FieldValue.Type> shardKeys = entityInformation.
+                    getShardKeys();
+            Map<String, FieldValue.Type> nonShardKeys = entityInformation.
+                    getNonShardKeys();
+
+            Map<String, String> entityShardMap = new LinkedHashMap<>();
+            shardKeys.forEach((k, v) -> entityShardMap.put(
+                    k.toLowerCase(), v.name().toLowerCase()));
+
+            Map<String, String> entityNonShardMap = new LinkedHashMap<>();
+            nonShardKeys.forEach((k, v) -> entityNonShardMap.put(
+                    k.toLowerCase(), v.name().toLowerCase()));
+
+            Map<String, String> entityOthersMap = new LinkedHashMap<>();
+            entityOthersMap.put(JSON_COLUMN.toLowerCase(), "json");
+
+            // convert maps to String
```

Comment:
why are you not using Map.equals?

Reply:
Map.equals also work. But anyway I need to convert maps to string to display proper error message. I'm just using string comparison. Default Map.toString() is not in proper format for our requirement